GVFS/GVFS.Common/Git/GitProcess.cs

```diff
@@ -164,6 +164,61 @@ public virtual void RevokeCredential(string repoUrl)
                 null);
         }
 
+        /// <summary>
+        /// Input for certificate credentials looks like
+        /// <code> protocol=cert
+        /// path=[http.sslCert value]
+        /// username =</code>
+        /// </summary>
+        public virtual bool TryGetCertificatePassword(
+            ITracer tracer,
+            string certificatePath,
+            out string password,
+            out string errorMessage)
+        {
+            password = null;
+            errorMessage = null;
+
+            using (ITracer activity = tracer.StartActivity("TryGetCertificatePassword", EventLevel.Informational))
+            {
+                Result gitCredentialOutput = this.InvokeGitAgainstDotGitFolder(
+                    "credential fill",
+                    stdin => stdin.Write("protocol=cert\npath=" + certificatePath + "\nusername=\n\n"),
+                    parseStdOutLine: null);
+
+                if (gitCredentialOutput.ExitCodeIsFailure)
+                {
+                    EventMetadata errorData = new EventMetadata();
+                    errorData.Add("CertificatePath", certificatePath);
+                    tracer.RelatedWarning(
+                        errorData,
+                        "Git could not get credentials: " + gitCredentialOutput.Errors,
+                        Keywords.Network | Keywords.Telemetry);
+                    errorMessage = gitCredentialOutput.Errors;
+
+                    return false;
+                }
+
+                password = ParseValue(gitCredentialOutput.Output, "password=");
+
+                bool success = password != null;

```

Comment:
Are empty passwords allowed?

Reply:
This is the same behavior as here:
https://github.com/Microsoft/VFSForGit/blob/7390eba79729141c9042b93750a108cdf4dafc2a/GVFS/GVFS.Common/Git/GitProcess.cs#L200

The idea is, that empty password will be an empty string, not a null.